### ISSUE TITLE
CRUD component crashes when trying to open the editor dialog on Safari

### ISSUE THREAD

**Issue by mlindfors** [vaadin/flow-components#1794]
### Description

Opening a CRUD's editor dialog causes the page to abruptly hit a severe error and reload.

### Expected outcome

The editor form dialog opens.

### Actual outcome

The browser tells about an error, and reloads the page.

### Steps to reproduce

1. Go to https://vaadin.com/components/vaadin-crud/java-examples with Safari
2. Look at the "Basic CRUD" example
3. Click at the button at the end of any row
4. Watch the browser reload the page

### Environment

- Vaadin 14.6.3, although I don't know what version the demos currently use. But happens on 14.6.3 as well.
- OS: macOS Catalina 10.15.7, also on iOS 14.6.
- Browser: Safari 14.1.1, or 14.6 on iOS.

### Browsers Affected

- [ ] Chrome
- [ ] Firefox
- [x] Safari
- [ ] Edge
- [ ] IE 11
- [x] iOS Safari
- [ ] Android Chrome

**Comment by vlukashov** [comment]
Most likely this is caused by a recent WebKit regression: https://bugs.webkit.org/show_bug.cgi?id=224408.
Safari 14.1 works fine whereas 14.1.1 has the issue.

**Comment by vlukashov** [comment]
Timebox 1d to investigate how feasible would it be to provide a workaround for the Safari issue.

**Comment by vlukashov** [comment]
The WebKit issue is closed / fixed on 2021-06-17. The fix may be coming in the next patch version of iOS / macOS.

ToDo: verify whether or not the issue remains in the next version of Safari (newer than 14.1.1 on macOS / 14.6 on iOS)

**Comment by web-padawan** [comment]
The logic that causes crash is [this line](https://github.com/vaadin/vaadin-crud/blob/6386536a1715b542ba987ae3609129aa6578cc95/src/vaadin-dialog-layout.js#L210).

The problem is that we teleport nodes from `vaadin-dialog-layour` including `<div id="editor">` and it includes `slot`.
So if we adjust the logic to not teleport the `slot` element but its content, then the crash does not happen:

```js
Array.from(this.childNodes).forEach((node) => {
  if (node.nodeType === Node.ELEMENT_NODE && node.id === 'editor') {
    this.$.dialog.$.overlay.$.content.appendChild(node.querySelector(':not(slot)'))
  } else {
    this.$.dialog.$.overlay.$.content.appendChild(node)
  }
});
```

This is enough to prevent crash in Safari 14.1.1 when using default editor position.

**Comment by web-padawan** [comment]
Note, the above code causes an error in Chrome on subsequent editing, so the workaround needs to be tweaked.

```
vaadin-dialog-layout.html.js:120 Uncaught TypeError: Failed to execute 'appendChild' on 'Node': parameter 1 is not of type 'Node'.
    at vaadin-dialog-layout.html.js:120
    at Array.forEach (<anonymous>)
    at HTMLElement._ensureChildren (vaadin-dialog-layout.html.js:118)
  ```

**Comment by silvan-lincan** [comment]
Hi, only wanted to mention that I just upgraded to iOS 14.7 and iPadOS 14.7 and the issue is gone, the Crud component works again.
Also, there is an update available for MacOS to 11.5, so I guess the Crud component will work also there. 
Thank you for the investigation.
